stonesoup/metricgenerator/base.py

```diff
@@ -30,6 +30,17 @@ def compute_metric(self, manager, **kwargs):
         """
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def generator_name(self) -> str:
+        """Unique identifier when accessing generated metrics from MultiManager.
+
+        Returns
+        -------
+        : unique identifier as string
+        """
+        raise NotImplementedError
+
```

Comment:
With that, some tests are failing, so we have a choice between removing the abstract property and include it in inherited classes **implicitly** or fix the following unittests unittests (which should be easy):

```
FAILED stonesoup/metricgenerator/tests/test_tables.py::test_siaptable - TypeError: Can't instantiate abstract class SIAPTableGenerator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_metric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_plottingmetric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_singletimemetric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_timerangemetric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_timerangeplottingmetric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
FAILED stonesoup/types/tests/test_metric.py::test_single_timeplottingmetric - TypeError: Can't instantiate abstract class temp_generator with abstract method generator_name
```

How what is your opinion?

Reply:
I think fixing the tests would be best, so `temp_generator` implements new abstract value. `temp_generator` probably doesn't need to be redefined in every test as well.

Reply:
Adapted